ci-operator/step-registry/rosa/sts/cluster/provision/rosa-sts-cluster-provision-commands.sh

```diff
@@ -131,26 +117,98 @@ if [[ "$DISABLE_WORKLOAD_MONITORING" == "true" ]]; then
 fi
 
 HYPERSHIFT_SWITCH=""
+SUBNET_ID_SWITCH=""
 if [[ "$HOSTED_CP" == "true" ]]; then
-  PUBLIC_SUBNET_ID=$(cat ${SHARED_DIR}/public_subnet_ids | tr -d "[']")
-  PRIVATE_SUBNET_ID=$(cat ${SHARED_DIR}/private_subnet_ids | tr -d "[']")
-  if [[ -z "${PUBLIC_SUBNET_ID}" ]] || [[ -z "${PRIVATE_SUBNET_ID}" ]]; then
-    echo -e "The public_subnet_id and the the privated_subnet_id are mandatory."
+  HYPERSHIFT_SWITCH="--hosted-cp"
+
+  PUBLIC_SUBNET_IDs=$(cat ${SHARED_DIR}/public_subnet_ids | tr -d "[']")
+  PRIVATE_SUBNET_IDs=$(cat ${SHARED_DIR}/private_subnet_ids | tr -d "[']")
+  if [[ -z "${PUBLIC_SUBNET_IDs}" ]] || [[ -z "${PRIVATE_SUBNET_IDs}" ]]; then
+    echo -e "The public_subnet_ids and the the privated_subnet_ids are mandatory."
```

Comment:
For the latest change last week, the hosted-cp cluster creation uses the byo oidc config as default and it needs to specify --oidc-config-id flag. And the classic flow is also supported, there is a hidden flag '--classic-oidc-config' for that. Detail see the fix PR of https://issues.redhat.com/browse/SDA-8696